pkg/pub_worker/lib/src/bin/dartdoc_wrapper.dart

```diff
@@ -22,75 +22,76 @@ Future<void> postProcessDartdoc({
         .list(recursive: true, followLinks: false)
         .whereType<File>();
-    await for (final file in originalFiles) {
-      if (cutoffTimestamp.isBefore(clock.now())) {
-        _log.warning(
-            'Cut-off timestamp reached during dartdoc archive building.');
-        break;
+    await for (final file in files) {
+      final suffix = file.path.substring(docDir.length + 1);
+      final targetFile = File(p.join(tmpOutDir, suffix));
+      await targetFile.parent.create(recursive: true);
+      final isDartDocSidebar =
+          file.path.endsWith('.html') && file.path.endsWith('-sidebar.html');
+      final isDartDocPage = file.path.endsWith('.html') && !isDartDocSidebar;
+      if (isDartDocPage) {
+        final page =
+            DartDocPage.parse(await file.readAsString(encoding: _utf8));
+        await targetFile.writeAsBytes(_jsonUtf8.encode(page.toJson()));
+      } else if (isDartDocSidebar) {
+        final sidebar =
+            DartDocSidebar.parse(await file.readAsString(encoding: _utf8));
+        await targetFile.writeAsBytes(_jsonUtf8.encode(sidebar.toJson()));
+      } else {
+        await file.copy(targetFile.path);
       }
-      // inside the archive prefix the name with <package>/version/
-      final relativePath = p.relative(file.path, from: originalDocDir.path);
-      final tarEntryPath = p.join(package, version, relativePath);
-      final data = await file.readAsBytes();
-      yield TarEntry.data(
-        TarHeader(
-          name: tarEntryPath,
-          size: data.length,
-        ),
-        data,
-      );
     }
-  }
+  }).whenComplete(() {
+    _log.info('Finished post-processing');
+  });
 
+  _log.info('Creating .tar.gz archive');
   final tmpTar = File(p.join(outputFolder, '_package.tar.gz'));
-  await _list()
-      .transform(tarWriter)
-      .transform(gzip.encoder)
-      .pipe(tmpTar.openWrite());
-  await tmpTar.rename(p.join(outputFolder, 'doc', 'package.tar.gz'));
+  final archiveFuture = Isolate.run(() async {
+    Stream<TarEntry> _list() async* {
+      final originalDocDir = Directory(docDir);
+      final originalFiles = originalDocDir
+          .list(recursive: true, followLinks: false)
+          .whereType<File>();
+      await for (final file in originalFiles) {
+        // inside the archive prefix the name with <package>/version/
+        final relativePath = p.relative(file.path, from: originalDocDir.path);
+        final tarEntryPath = p.join(package, version, relativePath);
+        final data = await file.readAsBytes();
+        yield TarEntry.data(
+          TarHeader(
+            name: tarEntryPath,
+            size: data.length,
+          ),
+          data,
+        );
+      }
+    }
+
+    await _list()
+        .transform(tarWriter)
+        .transform(gzip.encoder)
+        .pipe(tmpTar.openWrite());
+  }).whenComplete(() {
+    _log.info('Finished .tar.gz archive');
+  });
 
-  _log.info('Finished .tar.gz archive');
+  await Future.wait([
```

Comment:
What a shame `Isolate.run` doesn't support timeouts or some easy way to kill the isolate after a timeout...

Maybe we have to write that abstraction ourselves.

Reply:
Oh, I can add a timer that just kills the isolate from the inside... maybe in a follow-up PR.

Reply:
...Unless we have sync tasks blocking inside the isolate...

Reply:
Maybe we don't need this at all, as worst case the parent process will kill pana_wrapper after the set timeout.